File: 25/11/1578.cpp
#include <string>
#include <vector>
#include <iostream>

int minCost(std::string colors, std::vector<int> &neededTime)
{
    int n = colors.size();
    colors += '#';
    int m = 0, count = 0, ans = 0;
    for (int i = 0; i < n; ++i)
    {
        m = std::max(neededTime[i], m);
        count += neededTime[i];
        if (colors[i] == colors[i + 1])
        {
            continue;
        };
        ans += count - m;
        m = 0;
        count = 0;
    }
    return ans;
}

int main()
{
    std::string c = "abaac";
    std::vector<int> need = {1, 2, 3, 4, 5};
    std::cout<<minCost(c, need);
}

File: 25/11/go/1578.go
package L12

func minCost(colors string, neededTime []int) int {
	count := 0
	ans := 0
	m := 0
	for i := 0; i < len(colors); i++ {
		count += neededTime[i]
		m = max(m, neededTime[i])
		if i != len(colors)-1 && colors[i] == colors[i+1] {
			continue
		}
		ans += count - m
		m = 0
		count = 0
	}
	return ans
}
